content/tutorials/r-core/2-datacleaning/index.rmd

```diff
@@ -170,7 +170,7 @@ In this case, the missing value is the Age value in row 39. You know you have th
 ```{r}
 # Directly replace a missing value
 mydata <- mydata %>% 
-  mutate(Age = if_else(ID == 39, 30L, Age))
+  mutate(Age = if_else(ID == 39, as.integer(30), Age))
```

Comment:
Adding the suffix L to integers is a handy shortcut to specifying these numbers as ints, and is less wordy than calling `as.integer()` on a constant. Is there a style reason you've changed these? (As you can see my preference is to specify int constants with the L suffix but happy to switch if you have a particular motivation for this!)

Reply:
This line didn't run on my computer during the tutorial as written with `30L`, but `as.integer(30)` worked. (Unfortunately I hadn't noticed the issue prior to the tutorial when we shifted to `if_else()`, which is stricter about data types than `ifelse()`.) I'm not sure why it is; perhaps it has something to do with the fact that I had updated R and all my packages recently(?). Regardless, given that it didn't run, I thought it made sense to make the change so others don't experience the same issue!

Reply:
Huh! When I tested it just now on my machine, the `Age` column read in as double (which I suspected would happen), which means that neither of our versions should work because `if_else()` expects both to be double instead of int. We should probably either make it something like:

```suggestion
  mutate(Age = if_else(ID == 39, 30L, as.integer(Age)))
```

if we want to forcibly recode `Age` as int (it doesn't look like we do it earlier in the lesson).

Reply:
(broken into new comment to keep to one suggestion per comment)

Or we  could just not coerce things to int.
```suggestion
  mutate(Age = if_else(ID == 39, 30, Age))
```

We're not running into giant file size problems here that make it more important to coerce columns to int, so I'm ok with this too!

Reply:
I think I figured out what's going on!

If you load the data using `read_csv()` on line 72, `Age` reads in as double; however, if you load it using `load()` on line 79, `Age` reads in as integer. So: the `if_else()` line needs to be specified differently depending on the version you're working with.

I think that when I led the tutorial, I only ran the `read_csv()` line (which treating `Age` as double), which is why `30L` threw an error. Not a problem with my instance of R...a problem reflecting that I didn't overwrite the first version of the data (`read_csv()`) with the second version (`load()`).

I'm going to update the .rda version of the data so `Age` is double, and then we should be able to run the line you suggested (`mutate(Age = if_else(ID == 39, 30, Age))`), which is also my preferred way of editing this. :)

Reply:
Ooooh good catch! I didn't try reading in the .rda file so I definitely didn't see that. Your fix looks good :)

Reply:
Fixed! When you have a moment, could you please approve the PR? Thanks as always, Monica!